src/redismodule.h

```diff
@@ -257,6 +259,113 @@ typedef enum {
+    struct RedisModuleCommandArg *subargs;
+} RedisModuleCommandArgV1;
+
+#define RedisModuleCommandArg RedisModuleCommandArgV1
+
+typedef struct RedisModuleCommandHistoryEntry {
+    const char *since;
+    const char *changes;
+} RedisModuleCommandHistoryEntryV1;
+
+#define RedisModuleCommandHistoryEntry RedisModuleCommandHistoryEntryV1
+
+typedef struct RedisModuleCommandKeySpec {
+    uint64_t flags; /* REDISMODULE_CMD_KEY_* macros. */
+    RedisModuleKeySpecBeginSearchType begin_search_type;
+    union {
+        struct {
+            /* The index from which we start the search for keys */
+            int pos;
+        } index;
+        struct {
+            /* The keyword that indicates the beginning of key args */
+            const char *keyword;
+            /* An index in argv from which to start searching.
+             * Can be negative, which means start search from the end, in reverse
+             * (Example: -2 means to start in reverse from the panultimate arg) */
+            int startfrom;
+        } keyword;
+    } bs;
+    RedisModuleKeySpecFindKeysType find_keys_type;
+    union {
+        struct {
+            /* Index of the last key relative to the result of the begin search
+             * step. Can be negative, in which case it's not relative. -1
+             * indicating till the last argument, -2 one before the last and so
+             * on. */
+            int lastkey;
+            /* How many args should we skip after finding a key, in order to
+             * find the next one. */
+            int keystep;
+            /* If lastkey is -1, we use limit to stop the search by a factor. 0
+             * and 1 mean no limit. 2 means 1/2 of the remaining args, 3 means
+             * 1/3, and so on. */
+            int limit;
+        } range;
+        struct {
+            /* Index of the argument containing the number of keys to come
+             * relative to the result of the begin search step */
+            int keynumidx;
+            /* Index of the fist key. (Usually it's just after keynumidx, in
+             * which case it should be set to keynumidx + 1.) */
+            int firstkey;
+            /* How many args should we skip after finding a key, in order to
+             * find the next one, relative to the result of the begin search
+             * step. */
+            int keystep;
+        } keynum;
+    } fk;
+} RedisModuleCommandKeySpecV1;
+
+#define RedisModuleCommandKeySpec RedisModuleCommandKeySpecV1
+
+typedef struct RedisModuleCommandInfo {
+    const char *summary;          /* Summary of the command */
+    const char *complexity;       /* Complexity description */
+    const char *since;            /* Debut module version of the command */
+    RedisModuleCommandHistoryEntry *history; /* History */
+    /* An array of strings that are meant o be hints for clients/proxies
+     * regarding this command */
+    const char **hints;
```

Comment:
should be renamed  to `tips`

Reply:
Yes, when your tips PR is merged, I'll update this.

Reply:
PR merged